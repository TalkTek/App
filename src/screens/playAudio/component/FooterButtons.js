import React, { Component } from 'react'
import FunctionIcon from '../../../components/img/icon/XLIcon'
import { H5 } from '../../../components/text'
import {
  TouchableHighlight,
  View
} from 'react-native'
import { COLORS } from 'StyleConfig'

export class FooterButtons extends Component {
  style = {
    flex: 1,
    flexDirection: 'row',
    bottom: 10,
    borderTopWidth: 1,
    borderTopColor: COLORS.borderTop,
    height: 75,
    alignItems: 'center',
    justifyContent: 'space-around'
  }

  wrapperStyle = {
    alignItems: 'center'
  }
  
  render() {
    let { data } = this.props
    let lists = Object.values(data||{}).map((button, i) => {
      return (
        <TouchableHighlight
          transparent
          key={i}
          onPress={typeof button.func === 'function'? button.func: null}
          underlayColor="#fff"
        >
          <View style={this.wrapperStyle}>
            <FunctionIcon
              source={button.checkActive? button.active: button.notActive}
            />
            <H5 gray>
              {!isNaN(this.props[button.name])? this.props[button.name]: button.name}
            </H5>
          </View>
        </TouchableHighlight>
      )
    })
    return (<View style={this.style}>{lists}</View>)
  }
}

export default FooterButtons
