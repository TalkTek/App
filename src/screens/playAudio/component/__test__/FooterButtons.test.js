import React from 'react'
import { shallow } from 'enzyme'
import { FooterButtons } from '../FooterButtons'

describe('playAudio/component/FooterButtons test', () => {
  let data = {
    good: {
      notActive: require('../../assets/img/playAudio/good.png'),
      active: require('../../assets/img/playAudio/goodActive.png'),
      checkActive: 'audioIsGood',
      name: 'likeCounter',
      func: () => {}
    }
  }

  it('should render a <FooterButtons> tag', () => {
    const wrapper = shallow(<FooterButtons />)
    expect(wrapper.find('View').length).toEqual(1)
    expect(wrapper.find('TouchableHighlight').length).toEqual(0)
  })

  it('should render same length item', () => {
    const wrapper = shallow(<FooterButtons data={data} />)
    expect(wrapper.find('TouchableHighlight').length).toEqual(1)
  })
})