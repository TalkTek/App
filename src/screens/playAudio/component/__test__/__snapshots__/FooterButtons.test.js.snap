// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playAudio/component/FooterButtons test should render same length item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderTopColor": "rgb(224, 224, 224)",
      "borderTopWidth": 1,
      "bottom": 10,
      "flex": 1,
      "flexDirection": "row",
      "height": 75,
      "justifyContent": "space-around",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 40,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        fontSize={10}
        gray={true}
        style={
          Array [
            Object {
              "color": "rgb(158, 158, 158)",
              "fontSize": 10,
              "fontWeight": null,
            },
            undefined,
          ]
        }
      >
        likeCounter
      </Text>
    </View>
  </View>
</View>
`;
